account_invoice_pricelist/models/account_invoice.py

```diff
@@ -63,6 +63,13 @@ def _onchange_product_id_account_invoice_pricelist(self):
             product.price, product.taxes_id, self.invoice_line_tax_ids,
             self.company_id)
 
+    @api.onchange('uom_id')
+    def _onchange_uom_id(self):
+        if self.invoice_id.pricelist_id:
+            self._onchange_product_id_account_invoice_pricelist()
+        else:
```

Comment:
Is it normal not to call super always ?

Reply:
I added a new commit. I have tested with pricelist and normal price and seams to be working.